### ISSUE TITLE
Support endless range from Ruby 2.6

### ISSUE THREAD

**Issue by antonzaytsev** [rubocop-hq/rubocop#7159]
Latest version doesn't have support of added feature `endless ranges` in Ruby 2.6.

--------

## Expected behavior

In case of proper syntax for endless range, Rubocop should pass

## Actual behavior

It fails with error

```
Inspecting 1 file
An error occurred while Lint/DuplicatedKey cop was inspecting /Users/anton/code/rubocop-endless-range-issues/test.rb:3:7.
To see the complete backtrace run rubocop -d.
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Lint/DuplicatedKey cop was inspecting /Users/anton/code/rubocop-endless-range-issues/test.rb:3:7.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop-hq/rubocop/issues

Mention the following information in the issue report:
0.71.0 (using Parser 2.6.3.0, running on ruby 2.6.1 x86_64-darwin18)
```

## Steps to reproduce the problem

I created repo with 2 test files (ruby files and rubocop config).
https://github.com/antonzaytsev/rubocop-endless-range-issue.
Literally any code with endless range will break rubocop.
For example
```ruby
hash = {
  1...20 => true,
  20... => false
}
```

## RuboCop version

```
$ [bundle exec] rubocop -V
0.71.0 (using Parser 2.6.3.0, running on ruby 2.6.1 x86_64-darwin18)
```

**Comment by koic** [comment]
Thanks for the feedback. I opened a PR #7160.
